File: 백준/Silver/11723. 집합/집합.py
import sys
input = sys.stdin.readline

bitset = 0

for _ in range(int(input())):
    command = input().split()
    
    if command[0] == 'add':
        x = int(command[1])
        bitset |= (1 << x)
    elif command[0] == 'remove':
        x = int(command[1])
        bitset &= ~(1 << x)
    elif command[0] == 'check':
        x = int(command[1])
        print(1 if bitset & (1 << x) else 0)
    elif command[0] == 'toggle':
        x = int(command[1])
        bitset ^= (1 << x)
    elif command[0] == 'all':
        bitset = (1 << 21) - 1
    elif command[0] == 'empty':
        bitset = 0

File: 백준/Silver/11723. 집합/집합.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <math.h>
#include <limits.h>
#include <ctype.h>

#define TRUE 1
#define FALSE 0


int main(){
    int M, bit = 0;
    scanf("%d", &M);

    char command[10];
    int x;

    while(M--){
        scanf("%s", command);

        if(strcmp(command, "add") == 0){
            scanf("%d", &x);
            bit |= (1 << (x-1));
        }else if(strcmp(command, "remove") == 0){
            scanf("%d", &x);
            bit &= ~(1 << (x-1));
        }else if(strcmp(command, "check") == 0){
            scanf("%d", &x);
            if(bit & (1 << (x-1))){
                printf("1\n");
            }else{
                printf("0\n");
            }
        } else if(strcmp(command, "toggle") == 0){
            scanf("%d", &x);
            bit ^= (1 << (x-1));
        } else if(strcmp(command, "all") == 0){
            bit = (1 << 20) - 1;
        }else if(strcmp(command, "empty") == 0){
            bit = 0;
        }
    }


}